allennlp/data/batch.py

```diff
@@ -112,20 +110,17 @@ def as_tensor_dict(
             therefore corresponds to the number of `TokenIndexers` used to index the
             `TextField`.  Each `Field` class is responsible for batching its own output.
         """
-        if padding_lengths is None:
-            padding_lengths = defaultdict(dict)
+        padding_lengths = padding_lengths or defaultdict(dict)
         # First we need to decide _how much_ to pad.  To do that, we find the max length for all
         # relevant padding decisions from the instances themselves.  Then we check whether we were
         # given a max length for a particular field and padding key.  If we were, we use that
         # instead of the instance-based one.
         if verbose:
-            logger.info(
-                "Padding batch of size %d to lengths %s", len(self.instances), str(padding_lengths)
-            )
+            logger.info(f"Padding batch of size {len(self.instances)} to lengths {padding_lengths}")
             logger.info("Getting max lengths from instances")
         instance_padding_lengths = self.get_padding_lengths()
         if verbose:
-            logger.info("Instance max lengths: %s", str(instance_padding_lengths))
+            logger.info(f"Instance max lengths: {instance_padding_lengths}")
```

Comment:
This is a bit sad that python can't handle this, but if you format logging using the %s way, it is lazily evaluated - whereas if you do it in the new `f` way, it is evaluated immediately. 

See e.g:
https://stackoverflow.com/questions/34619790/pylint-message-logging-format-interpolation

Reply:
Do you think it's worth following pylint's suggestion here?  My vote would be to just use f strings, unless it's something we know would be expensive.

Reply:
Yeah, not sure. It does mean that we could have pretty slow code running for someone who has explicitly disabled it, but I don't know whether it matters in the grand scheme of things.